File: daily/rust/Dec-10-transpose-matrix.rs
impl Solution {
    pub fn transpose(matrix: Vec<Vec<i32>>) -> Vec<Vec<i32>> {
        let row = matrix.len();
        let col = matrix[0].len();
        let mut res = vec![vec![0; row]; col];

        for i in 0..col {
            for j in 0..row {
                res[i][j] = matrix[j][i];
            }
        }
        res
    }
}

File: daily/cpp/Dec-10-transpose-matrix.cpp
using namespace std;

class Solution {
 public:
  vector<vector<int>> transpose(vector<vector<int>>& matrix) {
    int row = matrix.size();
    int col = matrix[0].size();
    vector<vector<int>> res(col, vector<int>(row, 0));

    for (int i = 0; i < col; ++i) {
      for (int j = 0; j < row; ++j) {
        res[i][j] = res[j][i];
      }
    }

    return res;
  }
};
